### PR TITLE
docs: refactor Installation Guide

### LINKED ISSUE #5673: docs: refactor Installation Guide
### Improved or typo fixed.

Hi, in our `How to build` docs[1], we may need to improve or refactor it due to the following issues:

1. Step 1 says we need to install dependencies first, when we open that document, it links to another page under the Other category. If this step is necessary, maybe we could combine them into one document.
2. When we run commands according to the Install Dependencies document, ETCD will be installed and executed by "nohup etcd &", this way will not persist ETCD's data;
3. Also, after we install the dependencies, the OpenResty will be installed, too. And if we try to install by using RPM directly, an error will occur and it says we need to uninstall OpenResty, this is a little wired. This step is removed by https://github.com/apache/apisix/pull/5659 yesterday.

[1] https://apisix.apache.org/docs/apisix/how-to-build

### Solution

After the meeting, most of the attendees agree to refactor the Install Documentation :) Maybe we could have a structure first on the new docs?

### PR DESCRIPTION
**Pull request by tzssangglass**
Closes #5673

Signed-off-by: tzssangglass <tzssangglass@gmail.com>

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: docs/en/latest/how-to-build.md [hunk 1]
```diff
@@ -94,6 +95,23 @@ Please refer to: [Installing Apache APISIX with Helm Chart](https://github.com/a
...
+```shell
+ETCD_VERSION='3.4.13'
+wget https://github.com/etcd-io/etcd/releases/download/v${ETCD_VERSION}/etcd-v${ETCD_VERSION}-linux-amd64.tar.gz
+tar -xvf etcd-v${ETCD_VERSION}-linux-amd64.tar.gz && \
+    cd etcd-v${ETCD_VERSION}-linux-amd64 && \
+    sudo cp -a etcd etcdctl /usr/bin/
+nohup etcd &
+```
+
+Note: When ETCD is started with the command `nohup etcd &`, the data in ETCD is not stored persistently. For more information on ETCD, please refer to the [ETCD documentation](https://etcd.io/docs/).
```
Comment (spacewander) [inline docs/en/latest/how-to-build.md:113]: `When ETCD is started with the command `nohup etcd &`, the data in ETCD is not stored persistently.` This statement is incorrect. We can remove it as this is not a doc about etcd.

File: docs/en/latest/how-to-build.md [hunk 2]
```diff
@@ -94,6 +95,23 @@ Please refer to: [Installing Apache APISIX with Helm Chart](https://github.com/a
   make install
   ```
 
+## Step 2: Install ETCD
+
+This step is required when installing Apache APISIX via RPM or Docker.
```
Comment (spacewander) [inline docs/en/latest/how-to-build.md:100]: Maybe we have an independent doc for "install from source"?

File: docs/en/latest/how-to-build.md [hunk 3]
```diff
@@ -21,22 +21,29 @@ title: How to build Apache APISIX
...
-You can install Apache APISIX via RPM Repository, RPM package, Docker, Helm Chart, and source release package. Please choose one from the following options.
+If the official OpenResty repository is not installed yet, the following command will help you automatically install both the repositories of OpenResty and Apache APISIX.
 
-### Installation via RPM Repository(CentOS 7)
+```shell
+sudo yum install -y https://repos.apiseven.com/packages/centos/apache-apisix-repo-1.0-1.noarch.rpm
+```
 
-This installation method is suitable for CentOS 7. For now, the Apache APISIX RPM repository for CentOS 7 is already supported. Please run the following commands to install the repository and Apache APISIX.
+If the official OpenResty repository is installed, the following command will help you automatically install both the repositories of Apache APISIX.
```
Comment (juzhiyuan) [inline docs/en/latest/how-to-build.md:38]: > install both the repositories of Apache APISIX.

Both...and...

File: docs/zh/latest/how-to-build.md [hunk 4]
```diff
@@ -94,6 +95,23 @@ sudo yum install -y https://repos.apiseven.com/packages/centos/apache-apisix-rep
...
   ```
 
+## 步骤2：安装 ETCD
+
+通过 RPM 或者 Docker 安装 Apache APISIX 时，需要执行此步骤。
+
+你可以通过 Docker 或者二进制等方式安装 ETCD。以下命令通过二进制方式安装 ETCD。
+
+```shell
+ETCD_VERSION='3.4.13'
```
Comment (juzhiyuan) [inline docs/zh/latest/how-to-build.md:105]: Maybe add a `$` prefix is clearer in the multiple lines 😄

**Comment by juzhiyuan** [comment]
> docs: refactor Installation Guide

Hi @tzssangglass, I'm confused by this file 😂 Because this file's title is `How to build Apache APISIX`, it's more like used in a Development environment (including the Make Test part).

We have feedbacks that it seems that we mixed the Installation Guide with Developer Guides, when users want to install and use Apache APISIX, they only need to follow Step 1 ~ Step 3, right? And they could continue reading AdminAPI docs 😄 

Note: This comment is not related to this PR 🙈

**Comment by juzhiyuan** [comment]
Here has one PR about Make Test: https://github.com/apache/apisix/issues/5726

**Comment by tokers** [comment]
> > docs: refactor Installation Guide
> 
> 
> 
> Hi @tzssangglass, I'm confused by this file 😂 Because this file's title is `How to build Apache APISIX`, it's more like used in a Development environment (including the Make Test part).
> 
> 
> 
> We have feedbacks that it seems that we mixed the Installation Guide with Developer Guides, when users want to install and use Apache APISIX, they only need to follow Step 1 ~ Step 3, right? And they could continue reading AdminAPI docs 😄 
> 
> 
> 
> Note: This comment is not related to this PR 🙈 

It's my understanding that we can prepare two files, the one for installation and the other one for building from the source.

File: docs/en/latest/how-to-build.md (hunk 1 above)
Comment (tzssangglass) [reply docs/en/latest/how-to-build.md]: done

File: docs/en/latest/how-to-build.md (hunk 2 above)
Comment (tzssangglass) [reply docs/en/latest/how-to-build.md]: I didn't find this document, I've changed the expression.

File: docs/zh/latest/how-to-build.md (hunk 4 above)
Comment (tzssangglass) [reply docs/zh/latest/how-to-build.md:105]: done

**Comment by tzssangglass** [comment]
> It's my understanding that we can prepare two files, the one for installation and the other one for building from the source.

agree.

are we going to do it in this PR?

**Comment by juzhiyuan** [comment]
> are we going to do it in this PR?

How about doing it in another PR? Because this PR's destination is to refactor only Install part.

**Comment by tzssangglass** [comment]
> How about doing it in another PR? Because this PR's destination is to refactor only Install part.

I think so

**Comment by tokers** [comment]
> > How about doing it in another PR? Because this PR's destination is to refactor only Install part.
> 
> I think so

+1.

File: docs/zh/latest/how-to-build.md [hunk 5]
```diff
@@ -21,36 +21,37 @@ title: 如何构建 Apache APISIX
...
-
-## 步骤2：安装 Apache APISIX
+## 步骤1：安装 Apache APISIX
 
 你可以通过 RPM 仓库、RPM 包、Docker、Helm Chart、源码包等多种方式来安装 Apache APISIX。请在以下选项中选择其中一种执行。
 
 ### 通过 RPM 仓库安装（CentOS 7）
 
-这种安装方式适用于 CentOS 7 操作系统。Apache APISIX 已经支持适用于 CentOS 7 的 RPM 仓库。请运行以下命令安装 RPM 仓库和 Apache APISIX。
+这种安装方式适用于 CentOS 7 操作系统。
```
Comment (juzhiyuan) [inline docs/zh/latest/how-to-build.md:30]: Users may ask: how about other versions of CentOS?

File: docs/zh/latest/how-to-build.md [hunk 6]
```diff
@@ -21,36 +21,37 @@ title: 如何构建 Apache APISIX
...
+$ sudo yum install -y https://repos.apiseven.com/packages/centos/apache-apisix-repo-1.0-1.noarch.rpm
+```
 
-# Will show the existing apisix packages
-sudo yum --showduplicates list apisix
+如果已安装 OpenResty 的官方 RPM 仓库，请使用以下命令自动安装 Apache APISIX 的 RPM 仓库。
 
-# Will install the latest apisix package
-sudo yum install apisix
+```
```
Comment (juzhiyuan) [inline docs/zh/latest/how-to-build.md:40]: We'd better keep the same lang mark as other changes 😄

**Review by juzhiyuan — APPROVED** [review]
LGTM!

File: docs/zh/latest/how-to-build.md (hunk 5 above)
Comment (tzssangglass) [reply docs/zh/latest/how-to-build.md:30]: we currently only offer RPMs for Centos 7. In fact, AFAIK, Centos 7 is very common, Centos 6 and 8 are less common.

File: docs/en/latest/how-to-build.md [hunk 7]
```diff
@@ -66,34 +67,49 @@ Please refer to: [Installing Apache APISIX with Helm Chart](https://github.com/a
 1. Create a directory named `apisix-2.11.0`.
 
   ```shell
-  mkdir apisix-2.11.0
+  $ mkdir apisix-2.11.0
```
Comment (leslie-tsang) [inline docs/en/latest/how-to-build.md:70]:
```suggestion
  $ APISIX_VERSION='2.11.0'
  $ mkdir apisix-${APISIX_VERSION}
```
Would this be better ?

File: docs/en/latest/how-to-build.md [hunk 8]
```diff
@@ -66,34 +67,49 @@ Please refer to: [Installing Apache APISIX with Helm Chart](https://github.com/a
...
   ```
 
 2. Download Apache APISIX Release source package.
 
   ```shell
-  wget https://downloads.apache.org/apisix/2.11.0/apache-apisix-2.11.0-src.tgz
+  $ wget https://downloads.apache.org/apisix/2.11.0/apache-apisix-2.11.0-src.tgz
```
Comment (leslie-tsang) [inline docs/en/latest/how-to-build.md:76]: Ditto

File: docs/en/latest/how-to-build.md (hunk 7 above)
Comment (tzssangglass) [reply docs/en/latest/how-to-build.md:70]: done

File: docs/en/latest/how-to-build.md [hunk 9]
```diff
@@ -21,36 +21,37 @@ title: How to build Apache APISIX
...
-## Step 1: Install dependencies
-
-The Apache APISIX runtime environment requires dependencies on NGINX and etcd.
-
-Before installing Apache APISIX, please install dependencies according to the operating system you are using. We provide the dependencies installation instructions for **CentOS7**, **Fedora 31 & 32**, **Ubuntu 16.04 & 18.04**, **Debian 9 & 10**, and **MacOS**, please refer to [Install Dependencies](install-dependencies.md) for more details.
-
-## Step 2: Install Apache APISIX
+## Step 1: Install Apache APISIX
 
 You can install Apache APISIX via RPM Repository, RPM package, Docker, Helm Chart, and source release package. Please choose one from the following options.
```
Comment (spacewander) [inline docs/en/latest/how-to-build.md:26]:
```suggestion
You can install Apache APISIX via RPM Repository, Docker, Helm Chart, and source release package. Please choose one from the following options.
```

File: docs/en/latest/how-to-build.md [hunk 10]
```diff
@@ -21,36 +21,37 @@ title: How to build Apache APISIX
...
-This installation method is suitable for CentOS 7. For now, the Apache APISIX RPM repository for CentOS 7 is already supported. Please run the following commands to install the repository and Apache APISIX.
+This installation method is suitable for CentOS 7.
+
+If the official OpenResty repository is not installed yet, the following command will help you automatically install both the repositories of OpenResty and Apache APISIX.
 
 ```shell
-sudo yum-config-manager --add-repo https://repos.apiseven.com/packages/centos/apache-apisix.repo
-# View the information of the latest apisix package
-sudo yum info -y apisix
+$ sudo yum install -y https://repos.apiseven.com/packages/centos/apache-apisix-repo-1.0-1.noarch.rpm
```
Comment (spacewander) [inline docs/en/latest/how-to-build.md:35]: I think
`sudo yum install -y https://repos.apiseven.com/packages/centos/apache-apisix-repo-1.0-1.noarch.rpm` is the same as
`sudo yum-config-manager --add-repo https://repos.apiseven.com/packages/centos/apache-apisix.repo`?

@imjoey Please correct me if I am wrong.

File: docs/en/latest/how-to-build.md [hunk 11]
```diff
@@ -66,34 +67,50 @@ Please refer to: [Installing Apache APISIX with Helm Chart](https://github.com/a
...
-  mkdir apisix-2.11.0
+  $ APISIX_VERSION='2.11.0'
+  $ mkdir apisix-${APISIX_VERSION}
   ```
 
 2. Download Apache APISIX Release source package.
 
   ```shell
-  wget https://downloads.apache.org/apisix/2.11.0/apache-apisix-2.11.0-src.tgz
+  $ wget https://downloads.apache.org/apisix/2.11.0/apache-apisix-${APISIX_VERSION}-src.tgz
```
Comment (spacewander) [inline docs/en/latest/how-to-build.md:77]:
```suggestion
  $ wget https://downloads.apache.org/apisix/${APISIX_VERSION}/apache-apisix-${APISIX_VERSION}-src.tgz
```

File: docs/en/latest/how-to-build.md [hunk 12]
```diff
@@ -66,34 +67,50 @@ Please refer to: [Installing Apache APISIX with Helm Chart](https://github.com/a
...
   ```
 
   You can also download the Apache APISIX Release source package from the Apache APISIX website. The [Apache APISIX Official Website - Download Page](https://apisix.apache.org/downloads/) also provides source packages for Apache APISIX, APISIX Dashboard and APISIX Ingress Controller.
 
 3. Unzip the Apache APISIX Release source package.
 
   ```shell
-  tar zxvf apache-apisix-2.11.0-src.tgz -C apisix-2.11.0
+  $ tar zxvf apache-apisix-2.11.0-src.tgz -C apisix-2.11.0
```
Comment (spacewander) [inline docs/en/latest/how-to-build.md:85]: Why don't use `${APISIX_VERSION}`?

File: docs/en/latest/how-to-build.md [hunk 13]
```diff
@@ -66,34 +67,50 @@ Please refer to: [Installing Apache APISIX with Helm Chart](https://github.com/a
...
   ```
 
+## Step 2: Install ETCD
+
+This step is required if you have installed only Apache APISIX via RPM, Docker or source code but not ETCD.
+
+You can install ETCD via Docker or binary etc. The following command installs ETCD via binary.
+
+```shell
+$ ETCD_VERSION='3.4.13'
```
Comment (spacewander) [inline docs/en/latest/how-to-build.md:106]: We should use a command from install-dependencies.sh

File: docs/en/latest/how-to-build.md (hunk 10 above)
Comment (imjoey) [reply docs/en/latest/how-to-build.md:35]: > I think `sudo yum install -y https://repos.apiseven.com/packages/centos/apache-apisix-repo-1.0-1.noarch.rpm` is the same as `sudo yum-config-manager --add-repo https://repos.apiseven.com/packages/centos/apache-apisix.repo`?
> 
> @imjoey Please correct me if I am wrong.

@spacewander not exactly. Except for adding the apache-apisix.repo,  installing the rpm will also add the official openersty.repo.

File: docs/en/latest/how-to-build.md (hunk 9 above)
Comment (zaunist) [reply docs/en/latest/how-to-build.md:26]: Hi, @spacewander . I suggest Keep the `RPM package` description, some companies need install APISIX locally，so they will install APISIX via RPM package.

File: docs/en/latest/how-to-build.md (hunk 10 above)
Comment (spacewander) [reply docs/en/latest/how-to-build.md:35]: Thanks for your explanation. 😄

File: docs/en/latest/how-to-build.md (hunk 13 above)
Comment (tzssangglass) [reply docs/en/latest/how-to-build.md:106]: I have tried, but now install-dependencies.sh is not support to only install etcd.

File: docs/en/latest/how-to-build.md (hunk 12 above)
Comment (tzssangglass) [reply docs/en/latest/how-to-build.md]: done

File: docs/en/latest/how-to-build.md (hunk 11 above)
Comment (tzssangglass) [reply docs/en/latest/how-to-build.md]: done

File: docs/en/latest/how-to-build.md (hunk 9 above)
Comment (tzssangglass) [reply docs/en/latest/how-to-build.md]: hi @zaunist , RPM Repository and RPM package are same meanings.

**Review by leslie-tsang — APPROVED** [review]

File: docs/en/latest/how-to-build.md (hunk 9 above)
Comment (zaunist) [reply docs/en/latest/how-to-build.md]: > hi @zaunist , RPM Repository and RPM package are same meanings.

Got it.

**Review by zaunist — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Pull request merged**